core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -113,7 +114,8 @@ private String metadataFileLocation(Table table) {
   }
 
   private FileIO fileIO(Table table) {
-    if (table.io() instanceof HadoopConfigurable) {
+    if (HadoopDependency.isHadoopCommonOnClasspath(SerializableTable.class.getClassLoader())
```

Comment:
What happens when the hadoop is not on the classpath, but the table is `HadoopConfigurable`?

Reply:
No configuration will be passed through.